### ISSUE TITLE
Modify the report "projects" (add column urgency or due date)

### ISSUE THREAD

**Issue by bjce** [GothenburgBitFactory/taskwarrior#2526]
First a huge thanks for all the developpers involved in the development and maintaining of taskwarrior which is an great application

I just installed taskwarrior yesterday (I have version 2.5.3 on MacOS) and I really like the hierarchical view of the report `projects` (with projects > subprojects > subsubprojects etc...)

As indicated in the documentation ([https://taskwarrior.org/docs/report.html](url)), it is possible to modify see the structure of a report s.t.

```
$ task show report.minimal

Config Variable            Value
report.minimal.columns     id,project,tags.count,description.count
report.minimal.description Minimal details of tasks
report.minimal.filter      status:pending or status:waiting
report.minimal.labels      ID,Project,Tags,Description
report.minimal.sort        project+/,description+

```
and then modify it in the `.taskrc` file

However when I type 
```
 task show report.projects

Config Variable Value
No matching configuration variables.
```

is the report `projects` part of "Built-In Static Reports" and hence not-modyiable or is it possible to modify it and if yes how (e.g. add "urgency" or "due" column?

Many thanks in advance for your attention and your time

**Comment by tbabej** [comment]
Hi @bjce,

the `task projects` is indeed a static report and cannot be configured. I'm not sure what is the desired outcome here, however. The command shows the number of tasks per project, so it's not clear how we would map urgency or due date of individual tasks into it. Would you want to that column to display aggregate values (i.e. urgency of the highest urgency task, or average urgency?).

**Comment by bjce** [comment]
Dear @tbabej

Many thanks for your quick answer. Sorry if I was not clear

Let's say that the output of the command  `$ task projects` is:

```
Project                Tasks
---------------------- -----
Project1                  4
  subproject1_1           2
  subproject1_2           1
Project2                  4
  subproject2_1           2
  subproject2_2           1
    subproject2_2_1       1

```

The idea would be to have customizable additionnal columns such as:

```
Project                Urg   Due  Tasks
---------------------- ----- --- -----
Project1               1      2d    4    
  subproject1_1        10.1   1d    2
  subproject1_2        2      4d    1
Project2               4.2    3d    4
  subproject2_1        7.2    7d    2
  subproject2_2        12     6d    1
    subproject2_2_1    1      5d    1
```

Is is it more clear now?

**Comment by tbabej** [comment]
Yeah, makes sense! Though there are multiple options how to aggregate a column, so I guess we would have to support column modifiers like this:

`report.projects.columns=due.max,urgency.average,...`

**Comment by bjce** [comment]
@tbabej yes exactly that would be great! The goal would be to make report projects modifiable! I also underand if it is currently not possible to do it.
